lib/puppet/provider/reboot_notify/notify.rb

```diff
@@ -57,18 +65,22 @@ def self.post_resource_eval
       (current_time - v["updated"]) > Facter.value(:uptime_seconds)
     }
 
-    if not records.empty? then
+    unless records.empty?
       msg = ["System Reboot Required Because:"]
 
       records.each_pair do |k,v|
         # This is a fail safe for empty 'reasons'
-        records[k]['reason'] = 'modified' if ( records[k]['reason'].nil? or records[k]['reason'].empty? )
+        records[k]['reason'] = 'modified' if ( records[k]['reason'].nil? || records[k]['reason'].empty? )
         msg << ["  #{k} => #{v['reason']}"]
       end
 
       Puppet.notice(msg.join("\n"))
     end
 
-    File.open(target,'w'){|fh| fh.puts(PSON.pretty_generate(records))}
+    begin
+      File.open(target,'w'){|fh| fh.puts(JSON.pretty_generate(records))}
+    rescue
```

Comment:
See comment above about rescuing specific exceptions and including the specifics of that error in the error message.